### DISCUSSION TITLE
Possible regression since `1.12.0` on `BIGSERIAL` handling

### DISCUSSION THREAD

**Discussion by hugo-ricateau** [sqlalchemy/alembic#1519] [Usage Questions] [answered]
**Describe the bug**
Since `alembic==1.12.0`, auto-generation now detects a type change for the [`BIGSERIAL` receipt](https://github.com/sqlalchemy/sqlalchemy/discussions/9790#discussioncomment-5908349)
```
INFO  [alembic.autogenerate.compare] Detected type change from BIGINT() to BIGSERIAL() on '<table>.<column>'
```

**Expected behavior**
Not sure if it is a regression or a new feature; if not a bug, what is the expected way to handle it?

**To Reproduce**
This model
```python3
class BIGSERIAL(BigInteger):
    pass

@compiles(BIGSERIAL, postgresql.dialect.name)
def _serial(element, compiler, **kw):
    return 'BIGSERIAL'

class Table(Base):
    column = Column(BIGSERIAL, index=True, nullable=False, comment='Not a primary key')
    ...
```
were originally created as
```
op.create_table(
    'table',
    sa.Column('column', path.to.BIGSERIAL(), nullable=False),
    ...,
)
```
and since `alembic==1.12.0`, auto-generation now produces a
```
op.alter_column('table', 'column', existing_type=sa.BIGINT(), type_=path.to.BIGSERIAL(), existing_nullable=False)
```
So I assume it is an issue when detecting the existing type?

**Versions**
 - OS: Cross-platform
 - Python: `3.11.6`
 - Alembic: `>=1.12.0`
 - SQLAlchemy: `1.4.53`
 - Database: PostgreSQL
 - DBAPI: Psycopg

**Comment by zzzeek** [comment, marked as answer]
so BIGSERIAL is a pseudo-type in PostgreSQL which will come back in reflection as BIGINT.    This is a custom type you're adding, so to have alembic ignore this you would need to write a [custom rule to compare types](https://alembic.sqlalchemy.org/en/latest/autogenerate.html#comparing-types) 

```
def my_compare_type(context, inspected_column,
            metadata_column, inspected_type, metadata_type):
    if isinstance(inspected_type, BIGINT):
        return False
    return None

context.configure(
    # ...
    compare_type = my_compare_type
)
```

**Reply by hugo-ricateau** [reply]
Indeed, I missed it:

> **Changed in version 1.12.0:** The default value of [`EnvironmentContext.configure.compare_type`](https://alembic.sqlalchemy.org/en/latest/api/runtime.html#alembic.runtime.environment.EnvironmentContext.configure.params.compare_type) has been changed to `True`.

Many thanks for your quick and very complete answer.
